cli/azd/pkg/pipeline/azdo_provider.go

```diff
@@ -702,17 +701,59 @@ func (p *AzdoCiProvider) credentialOptions(
 	repoDetails *gitRepositoryDetails,
 	infraOptions provisioning.Options,
 	authType PipelineAuthType,
-) *CredentialOptions {
-	if authType == "" || authType == AuthTypeClientCredentials {
+	credentials *azcli.AzureCredentials,
+) (*CredentialOptions, error) {
+	// Default auth type to client-credentials for terraform
+	if infraOptions.Provider == provisioning.Terraform && authType == "" {
+		authType = AuthTypeClientCredentials
+	}
+
+	if authType == AuthTypeClientCredentials {
 		return &CredentialOptions{
 			EnableClientCredentials: true,
+		}, nil
+	}
+
+	// If not specified default to federated credentials
+	if authType == "" || authType == AuthTypeFederated {
+		p.credentials = credentials
+		details := repoDetails.details.(*AzdoRepositoryDetails)
+		org, _, err := azdo.EnsureOrgNameExists(ctx, p.envManager, p.Env, p.console)
+		if err != nil {
+			return nil, err
+		}
+		pat, _, err := azdo.EnsurePatExists(ctx, p.Env, p.console)
```

Comment:
Does azdo still require a PAT? No other auth options that can consume some Azure credential?

Reply:
We could re-use azd-auth, but that would limit the user running azd to use the AzureDevOps linked to the Azure Tenant (if any).
Right now, by using a PAT, you can run `azd pipeline config` for any Azure DevOps Organization (regardless of what ADO is related to the same tenant where you are deploying your project)